src/lib/progress.h

```diff
@@ -0,0 +1,65 @@
+#ifndef __PROGRESS__
+#define __PROGRESS__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct step {
+	unsigned int current;
+	unsigned int total;
+	char *description;
+};
+
+/*
+ * available formats the output can be printed
+ */
+enum progress_formats {
+	standard,
+	json
+};
+
+/*
+ * Sets the format the progress functions should use to
+ * print their output.
+ */
+void set_progress_format(enum progress_formats);
+
+/*
+ * Initializes a process (a series of steps to complete a given task),
+ * this function should be used first so progress in steps is reported
+ * correctly.
+ */
+void init_steps(char *, int);
+
+/*
+ * Finishes the reporting of a process (as defined above). If the
+ * format used, needs a special closing statement, this function will
+ * print it.
+ */
+void finish_steps(char *);
+
+/*
+ * Sets a step of the process to be tracked/reported.
+ */
+void set_step(unsigned int, char *);
```

Comment:
add a prefix to all functions in this file. Like progress_ It's way easier to find where a function come from.

Reply:
ok, I'll do that